app/enterprise/0.33-x/developer-portal/best-practice.md

```diff
@@ -0,0 +1,204 @@
+│       └── vitals.js
+└── templates
+    ├── pages
+    │   ├── 404.hbs
+    │   ├── about.hbs
+    │   ├── dashboard.hbs
+    │   ├── documentation
+    │   │   ├── api1.hbs
+    │   │   ├── api2.hbs
+    │   │   └── loader.hbs
+    │   ├── guides
+    │   │   ├── 5-minute-quickstart.hbs
+    │   │   ├── index.hbs
+    │   │   ├── kong-architecture-overview.hbs
+    │   │   ├── kong-ee-introduction.hbs
+    │   │   ├── kong-vitals.hbs
+    │   │   └── uploading-spec.hbs
+    │   ├── index.hbs
+    │   ├── unauthenticated
+    │   │   ├── 404.hbs
+    │   │   ├── index.hbs
+    │   │   ├── login.hbs
+    │   │   ├── register.hbs
+    │   │   └── unauthorized.hbs
+    │   └── user.hbs
+    ├── partials
+    │   ├── custom-js.hbs
+    │   ├── footer.hbs
+    │   ├── header.hbs
+    │   ├── kong-docs-sidebar.hbs
+    │   ├── layout.hbs
+    │   ├── sidebar-spec.hbs
+    │   ├── sidebar.hbs
+    │   ├── spec-dropdown.hbs
+    │   ├── spec-renderer.hbs
+    │   └── unauthenticated
+    │       ├── auth-js.hbs
+    │       ├── code-snippet-languages.hbs
+    │       ├── custom-css.hbs
+    │       ├── footer.hbs
+    │       ├── header.hbs
+    │       ├── layout.hbs
+    │       ├── login-actions.hbs
+    │       └── title.hbs
+    └── specs
+        ├── admin.json
+        ├── files.yaml
+        └── vitals.yaml
+```
+
+When the file structure is setup, you can make use of [GitHub Webhooks](https://developer.github.com/webhooks/creating/) and every time you merge a change into the Master branch, it triggers a push of files to the portal. This allows your developers to work in an environment they are familiar with, and maintain a history of changes to your portal. It also means you can easily roll back to a previous version.
+
+## <a name=partial></a>Using Effective Partial Usage
+
+Partials in the portal allow you to create reusable elements of a page. A page will traditionally be made up of a group of partials. For example, if you wanted to create a page announcing a new product, you would probably want to include things such as the header, footer, sidebar etc. Each of those elements are saved as partials. For more information about partials, please refer to [our documentation](https://docs.konghq.com/enterprise/0.33-x/developer-portal/file-management/#partials).
+
+partials allow you to control what is on every page.n In order for your portal to load quickly, it is vital to keep these files as small as possible. The general rule is partials should be around 500 kB. They should never be larger than 1 MB.
+
+In order to achieve this, make sure to not do more than what is necessary for every partial. If something is not needed on every page, avoid placing it in a global file and instead, pull it into it's own partial.
+
+
+## <a name=avoid></a>Practices to Avoid
+
+You should never, under any circumstances store images or other media files in your Kong database. We suggest using a Content Delivery Network (CDN) or keeping them in an S3 bucket.
```

Comment:
This needs clarification. For example, is it OK to include small SVG files in Kong Dev Portal files? What about in-line SVG code?

Reply:
Note that the existing Dev Portal docs https://docs.konghq.com/enterprise/0.33-x/developer-portal/customization/#adding-image-video-or-other-file-types-to-a-page somewhat conflict with this advice - the existing docs discus including in-line SVGs.

Reply:
To be pedantic, when you add an in-line SVG to a page or a partial, then store that page or partial in the Kong, you have in effect stored a graphic file in Kong's database. If you made a partial that was _only_ an in-line SVG, then you are pretty clearly storing a graphic in Kong's database.

When you say "You should never, under any circumstances store images or other media files in your Kong database.", that conflicts with the advice on the other page "To add images, videos, and other file types to your pages, you must either add them inline (e.g. inline SVG)" - if you do the latter, you are violating the former. 

Additionally, with the exception of in-line SVGs, I don't believe it is even possible to upload graphic or video files to Kong - so, advising against it is not necessary. 

I'd suggest you resolve this by simply dropping these two sentences - this topic is already covered here https://docs.konghq.com/enterprise/0.33-x/developer-portal/customization/#adding-image-video-or-other-file-types-to-a-page - and if you want to clarify that, it'd be better to do so by improving that page, rather than "splitting" similar advice into two different places.